### ISSUE TITLE
Add memory tests for Firefox and with console open

### ISSUE THREAD

**Issue by jeswr** [eyereasoner/eye-js#1068]
To catch regressions once https://github.com/eyereasoner/eye-js/issues/1066#issuecomment-2094962853 is resolved.

**Comment by jeswr** [comment]
Agent-assisted triage for @jeswr — status check: this remains blocked. memlab still has no Firefox support (facebook/memlab#32 open), and the Firefox bug behind #1066 (bugzilla 1895309) was resolved as a duplicate of bugzilla 1864296, which is still open/NEW upstream — so a Firefox repeated-instantiation memory test would fail today through no fault of eye-js. When 1864296 lands, the practical implementation is a playwright-firefox (or selenium) loop harness against __test_utils__/browser.html asserting N instantiations complete without OOM (Firefox exposes no performance.memory, so assert absence-of-failure rather than heap deltas), plus a Chromium variant launched with devtools open for the console-open case. Keeping on backlog until the upstream fix.

**Comment by jeswr** [comment]
Closing this after re-verifying the upstream state: the test we want here is blocked on upstream work in two places.

**What this test needs**

This issue tracks adding an automated memory-regression test for Firefox (including the "DevTools/console open" case) to guard against regressions of the leak investigated in #1066.

**Upstream status**

1. **memlab has no Firefox support — this is the true blocker for adding the test.** Our existing memory tests use `@memlab/api`, which drives Chromium via Puppeteer and analyzes V8-format heap snapshots (Chromium / Node.js / Electron / Hermes only). Firefox/SpiderMonkey does not produce compatible snapshots, so there is currently no way to build this test on our existing harness. The closest upstream tracking issue is facebook/memlab#32 (still open); no dedicated Firefox-support issue exists there today.

2. **The underlying Firefox bug is properly tracked upstream.** The OOM from #1066 was reported via emscripten-core/emscripten#21892 and then [Bugzilla 1895309](https://bugzilla.mozilla.org/show_bug.cgi?id=1895309), which Mozilla resolved as a duplicate of [Bugzilla 1864296](https://bugzilla.mozilla.org/show_bug.cgi?id=1864296) (still open, status NEW, P3). I re-checked 1864296 and it captures our scenario precisely: repeatedly instantiating dynamic WebAssembly modules and dropping the references — exactly what eye-js does, booting a fresh swipl-wasm module per reasoner call — leaks JIT executable memory *when DevTools is open*, because DevTools' Debugger keeps the dead `Instance` objects alive (via `SourceActor`/`Debugger.Source` retention), until the 2 GiB executable-memory limit is hit ("failed to allocate executable memory for module"). Mozilla engineers have reproduced and root-caused it, so no additional Bugzilla report is needed from our side.

**Test design once unblocked**

When Firefox memory tooling exists, the test should mirror our existing Chromium memlab scenario: load the browser bundle in Firefox with DevTools attached, run repeated `n3reasoner` invocations, and assert that retained WASM `Instance` objects / executable memory do not grow monotonically across iterations (no OOM after N instantiations, both with and without the console open).

**Status**

Closing as *not planned*; this should be reopened once memlab (or an equivalent harness) gains Firefox support and/or Bugzilla 1864296 is resolved.

---
*This issue was triaged and closed with agent assistance on behalf of @jeswr.*
